docs/MultisiteExamples.md

```diff
@@ -0,0 +1,274 @@
+from all cloud providers (e.g. [GCP](https://cloud.google.com/solutions/scalable-and-resilient-apps),
+[AWS](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/#Availability_Zones),
+[Azure](https://docs.microsoft.com/en-us/azure/availability-zones/az-overview#availability-zones)).
+In a private datacenter, multisite indexer clusters can be used to support the loss of a room or rack.
+In the case of dedicated hardware with local storage used for Splunk (or various datastores),
+a multisite indexer cluster allows to support regular maintenance (e.g. OS upgrades) in the case of
+multiple indexer pods scheduled on the same host.
+
+## Multipart IndexerCluster
+
+Description: connect multiple IndexerCluster resources, each constrained to run within a dedicated zone and
+configured with a hardcoded site.
+
+Advantages:
+- some operations are performed per site which mitigates the risk of impact on the whole cluster (e.g. Splunk upgrades, scaling up resources)
+- specific indexer services are created per site allowing to send events to the indexers located in the same zone, avoiding possible cost of cross-zone traffic. Indexer discovery from cluster-master can do this for forwarders, but this solution also covers http/HEC traffic
+
+#### Deploy the cluster-master
+
+Note: the image version is defined in these resources as this allows to control the upgrade cycle 
+
+```yaml
+cat <<EOF | kubectl apply -f -
+---
+kind: IndexerCluster
+metadata:
+  name: example
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  replicas: 0
+  image: "splunk/splunk:8.0.4"
+  defaults: |-
+    splunk:
+      site: site1
+      multisite_master: localhost
+      all_sites: site1,site2,site3
+      multisite_replication_factor_origin: 1
+      multisite_replication_factor_total: 2
+      multisite_search_factor_origin: 1
+      multisite_search_factor_total: 2
+      idxc:
+        search_factor: 2
+        replication_factor: 2
+      # Apps defined here are deployed to the indexers of all the sites
+      apps_location:
+        - "https://example.com/splunk-apps/app3.tgz"
+EOF
+```
+
+#### Deploy the indexer sites
+
+Site / zone mapping must be adapted for each site.
+
+```yaml
+cat <<EOF | kubectl apply -f -
+---
+kind: IndexerCluster
+metadata:
+  name: example-site1
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  replicas: 2
+  image: "splunk/splunk:8.0.4"
+  indexerClusterRef:
+    name: example
+  defaults: |-
+    splunk:
+      multisite_master: splunk-example-cluster-master-service
+      site: site1
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: failure-domain.beta.kubernetes.io/zone
+            operator: In
+            values:
+            - zone-1a
+EOF
+```
+
+## Single IndexerCluster resource and Downward API
```

Comment:
The Downward API approach seems a lot more complicated to me. I question whether it's a good idea to document here versus having just one recommended & supported way to setup multisite?

Reply:
I agree that it is more complicated, I just wanted to provide something as complete as possible and throught some user might prefer to operate a single resource, but I don't see any problem to remove it. You could also remove it after merging the PR if you want to keep it in the history.

This type of setup could be integrated within the operator, automatically configuring the downward API, directly using the node name as a variable instead of doing two API calls with ansible lookups. Then a new parameter could be passed to ansible to provide the mapping between zones and sites, and the retrieval of the site could be integrated within splunk-ansible. That could reduce a lot the complexity.

This would have required heavier changes and been more intrusive, maybe adding a new parameter to the CRD, adding a module in the docker image, modifying ansible-splunk, so I didn't feel comfortable to propose this as an external contributor. This also wouldn't have solved my problem of defining different specs for cluster-master and indexers (local SSD storage).

The only difficulty which I am currently experiencing with the dedicated IndexerCluster resource per site is related to autoscaling. When adding indexers, the number of indexers doesn't necessarily remain balanced across sites. When scaling down, it currently removes indexers from several sites in parallel. I didn't expect issues as this is a graceful offline, but I'm actually experiencing troubles with buckets not searchable (which I plan to discuss together with Splunk PS). A setup using a single resource and downward API would of course solve this, but there might be other options in the operator to control this.